code/missionui/missionshipchoice.cpp

```diff
@@ -1853,6 +1853,25 @@ bool is_weapon_carried(int weapon_index)
 	return false;
 }
 
+bool is_a_weapon_slot_empty()
+{
+	for (int slot = 0; slot < MAX_WING_BLOCKS*MAX_WING_SLOTS; slot++)
+	{
+		// a ship must exist in this slot
+		if (Wss_slots[slot].ship_class >= 0)
+		{
+			for (int bank = 0; bank < MAX_SHIP_WEAPONS; bank++)
```

Comment:
If the suggested change is not desired then please suppress it with `// NOLINT(modernize-loop-convert)`.

Reply:
Ok.